Rename az_iot_hub_client_properties to az_iot_message_properties (#1124)
Azure/azure-sdk-for-c: CHANGELOG.md

- Add `az_json_writer_append_json_text()` to support appending existing JSON with the JSON writer.

> - Add supported, system properties for IoT Hub messages to az_iot_common.h

Reviewer: nit: back-ticks and period.
```suggestion
- Add support for system properties for IoT Hub messages to `az_iot_common.h`.
```

> - Rename and move `az_iot_hub_client_properties` (in az_iot_hub_client.h) to `az_iot_message_properties` (in az_iot_common.h)

Reviewer: This is a breaking change, move it to that section.

nit: Add back-ticks to filenames.

Also some re-wording suggestion.

```suggestion
- Rename `az_iot_hub_client_properties` to `az_iot_message_properties` and move it from `az_iot_hub_client.h` to `az_iot_common.h`.
```

Author: ahh thank you didn't see the breaking change section
